### PR TITLE
ProgrammingGuideClassesAndStructsFinalizers

### PR DESCRIPTION
**Pull request by deep-outcome**
## Summary

I was about to fix physical location names under _dotnet\docs\docs\csharp\programming-guide\classes-and-structs_ with _finalizer_ instead of _destructor_. But then I found that I am unable to fix references (links) in other repositories simply because they do not contain them although pages do.

For instance https://docs.microsoft.com/en-us/dotnet/api/system.idisposable.dispose?view=net-5.0 contains https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/destructors. This is beyond my knowledge.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=26101) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=26101) before we can accept your contribution.<br/><br/>:x: Uzivatel919 [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=26101)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=26101) it.</sub>

**Comment by BillWagner** [comment]
Thanks for this @Uzivatel919 

First, you'll need to address the merge conflicts before this goes further. That way we can see the build results.

I skimmed the changes, and the text changes are great. 

The file name changes may be more work than the benefit. (At any rate, that's what I thought when the ECMA committee formally adopted a preference for "finalizer". One option is to revert the file name changes.

The other option is to add a redirect in the [redirection file](https://github.com/dotnet/docs/blob/main/.openpublishing.redirection.json).  Note that entries are sorted alphabetically by source path. That minimizes merge conflicts when multiple authors have PRs open to move files.

I'll watch for updates and merge this when it's ready.

**Comment by Uzivatel919** [comment]
Added redirect, conflict solved.

File: .openpublishing.redirection.json [hunk 1]
```diff
@@ -1904,6 +1904,14 @@
...
             "redirect_url": "/dotnet/csharp/fundamentals/types/classes"
         },
+	{
+            "source_path": "docs/csharp/programming-guide/classes-and-structs/finalizers.md",
+            "redirect_url": "/dotnet/csharp/programming-guide/classes-and-structs/destructors"
+        },
+        {
+            "source_path": "docs/csharp/programming-guide/classes-and-structs/finalizers.md",
+            "redirect_url": "/docs/csharp/programming-guide/classes-and-structs/destructors.md"
+        },
```
Comment (BillWagner) [inline .openpublishing.redirection.json:1914]: These lines should be removed (it looks like they were added twice)

**Review by BillWagner — COMMENTED** [review]
Thanks @Uzivatel919 

I have one comment that would resolve the build errors. Once the build is clean, I'll :shipit:

Thanks for helping us.

File: .openpublishing.redirection.json (hunk 1 above)
Comment (Uzivatel919) [reply .openpublishing.redirection.json]: I removed it but there is still some error. As I suggested before but obviously not enough clearly.

File: .openpublishing.redirection.json [hunk 2]
```diff
@@ -1904,6 +1904,10 @@
             "source_path": "docs/csharp/programming-guide/classes-and-structs/classes.md",
             "redirect_url": "/dotnet/csharp/fundamentals/types/classes"
         },
+	{
+            "source_path": "docs/csharp/programming-guide/classes-and-structs/finalizers.md",
+            "redirect_url": "/dotnet/csharp/programming-guide/classes-and-structs/destructors"
```
Comment (gewarren) [inline .openpublishing.redirection.json:1909]: `destructors` should be the `source_path` file, and `finalizers` should be the `redirect_url` file.

Comment (Uzivatel919) [reply .openpublishing.redirection.json]: @gewarren, unfortunately it seems not.

File: .openpublishing.redirection.json [hunk 3]
```diff
@@ -1904,6 +1904,10 @@
             "source_path": "docs/csharp/programming-guide/classes-and-structs/classes.md",
             "redirect_url": "/dotnet/csharp/fundamentals/types/classes"
         },
+        {
+            "source_path": "/dotnet/csharp/programming-guide/classes-and-structs/destructors",
```
Comment (gewarren) [inline .openpublishing.redirection.json:1908]:
```suggestion
            "source_path": "docs/csharp/programming-guide/classes-and-structs/destructors.md",
```

File: .openpublishing.redirection.json [hunk 4]
```diff
@@ -1904,6 +1904,10 @@
...
+            "redirect_url": "docs/csharp/programming-guide/classes-and-structs/finalizers.md"
```
Comment (gewarren) [inline .openpublishing.redirection.json:1909]:
```suggestion
            "redirect_url": "/dotnet/csharp/programming-guide/classes-and-structs/finalizers"
```

File: .openpublishing.redirection.json [hunk 5]
```diff
@@ -1904,6 +1904,10 @@
...
+        },
```
Comment (BillWagner) [inline .openpublishing.redirection.json:1910]: It should look like this:

```suggestion
        {
            "source_path": "/docs/csharp/programming-guide/classes-and-structs/destructors.md",
            "redirect_url": "/dotnet/csharp/programming-guide/classes-and-structs/finalizers"
        },
```

**Review by gewarren — COMMENTED** [review]
Sorry, I didn't mean switch the entire paths, just the file names. You're redirecting from destructors to finalizers.

File: .openpublishing.redirection.json (hunk 5 above)
Comment (gewarren) [reply .openpublishing.redirection.json:1910]: Except without the leading forward slash on line 1908.

Comment (BillWagner) [reply .openpublishing.redirection.json:1910]: Fixed. Great :eyes: @gewarren

**Review by BillWagner — APPROVED** [review]
Thanks @Uzivatel919 

Now that the build is clean, I'll :shipit:

You should see the changes on our live site in the next day or two, on our regular publishing cycle.

**Pull request merged**